app/controllers/panel_controller.rb

```diff
@@ -46,4 +53,22 @@ def generate_db_token
       replica: 2,
     }
   end
+
+  def anonymize_user
+    user_id = params.require(:userId)
+    user = User.find(user_id)
+    user.update!(
```

Comment:
If this `update!` fails, Rails will throw a hard error. This hard error will render an HTML error page, which JSON endpoints cannot handle very well.

A better pattern is to use `user_changed = user.update(...)`: When you use `update` method without the `!`, it will return a boolean indicating whether or not the validations that run while saving are successful.
Then you can react like `render json: { success: user_changed }`, and if `user_changed` is false you could even include `user.errors` to communicate the validation errors as a "clean" JSON to the frontend